### ISSUE TITLE
Allow to specify message encodings

### ISSUE THREAD

**Issue by sharok** [sketch7/signalr-client#21]
First of all, thanks for the library. SingalR supports binary data, but I can't see how I can enable it in the your library.
`.withHubProtocol(new signalR.protocols.msgpack.MessagePackHubProtocol())`

I think the lib should allow user to setup it.

**Comment by claylaut** [comment]
@sharok Thanks, please star my project if you liked it :) 

Nice catch, cause I've exposed the options but protocol is not exposed in `IHttpConnectionOptions`. 

I will do a fix for this.

**Comment by claylaut** [comment]
@sharok I've published a fix for this with version `0.7.1`
Now `ConnectionOptions` has optional property `protocol`.

example:
```
hubFactory.create(
			{
				key: "hero",
				endpointUri: "/real-time/hero",
				protocol: xxxx
			}
		);
```
